### PR TITLE
NPM 3 support!

### PR DESCRIPTION
**Pull request by rcorral**

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @MatApple

**Comment by MatApple** [comment]
Awesome! Nice @rcorral

File: package.json [hunk 1]
```diff
@@ -98,7 +97,9 @@
   },
   "scripts": {
     "clean": "rm -rf dist",
+    "prebuild-assets": "npm install compression-webpack-plugin@latest",
     "build-assets": "npm run clean && NODE_ENV=production webpack --config ./webpack/webpack.production.babel.js",
+    "postbuild-assets": "rm -rf node_modules/compression-webpack-plugin",
```
Comment (rcorral) [inline package.json]: These are needed because this node module doesn't play well with shrinkwrap

File: src/js/components/SidePanelContents.js [hunk 2]
```diff
@@ -21,7 +21,7 @@ const METHODS_TO_BIND = [
 ];
 
 class SidePanelContents
-  extends mixin(InternalStorageMixin, TabsMixin, StoreMixin) {
```
Comment (Poltergeist) [inline src/js/components/SidePanelContents.js]: This change is probably an accident, right?

**Comment by Poltergeist** [comment]
The `npm-shrinkwrap.json` is empty, this is why velocity is failing.

File: src/js/components/SidePanelContents.js (hunk 2 above)
Comment (rcorral) [inline src/js/components/SidePanelContents.js]: nah, eslint complains otherwise.

**Comment by jfurrow** [comment]
Nice work on hunting down all these caveats to get the build passing!

**Pull request merged**